apps/testrunner/actions/process_results.py

```diff
@@ -11,18 +13,100 @@
+            'query_time': 0.0,
+            'query_master': 0,
+            'query_slave': 0,
+            'memc_time': 0.0,
+            'memc_misses': 0,
+            'memc_hits': 0,
+            'memc_dupes': 0
+        }
+
+        for line in raw_data.splitlines():
+            match = self.__PROFILER_REGEXP.match(line)
+
+            if match is not None:
+                (percent, time, count, name) = match.groups()
+
+                percent = float(percent.strip('%'))
+                time = float(time)
+                count = int(count)
+
+                if name == '-total':
+                    metrics['server_time'] += time
+                elif name == 'DatabaseBase::query' or name == 'DatabaseBase::query-master':
+                    is_master = name == 'DatabaseBase::query-master'
+                    metrics['query_time'] += time
+
+                    if is_master:
+                        metrics['query_master'] += count
+                    else:
+                        metrics['query_slave'] += count
+                elif name == 'MWMemcached::get':
+                    metrics['memc_time'] += time
+                else:
+                    match = re.match(self.__MEMCACHE_REGEXP, name)
+
+                    if match is not None:
+                        memc_type = match.group(1)
+                        if memc_type == 'HIT':
+                            metrics['memc_hits'] += count
+                        elif memc_type == 'MISS':
+                            metrics['memc_misses'] += count
+                        elif memc_type == 'DUPE':
+                            metrics['memc_dupes'] += count
+        return metrics
+
+    @staticmethod
+    def _calculate_stats(values):
+        return {
+            'count': int(len(values)),
+            '5th_percentile': float(numpy.percentile(values, 5)),
+            '50th_percentile': float(numpy.percentile(values, 50)),
+            '90th_percentile': float(numpy.percentile(values, 90)),
+            '95th_percentile': float(numpy.percentile(values, 95)),
+            'std': float(numpy.std(values)),
+            'mean': float(numpy.mean(values)),
+            'median': float(numpy.median(values)),
+            'lowest': float(numpy.min(values)),
+            'highest': float(numpy.max(values)),
         }
 
+    def run(self):
+        logger.info('Starting processing results')
+
+        response_times = [
+            float(response.headers['X-Backend-Response-Time']) for response in self.params['results']['http_get']
+        ]
+
+        self.result['response_time'] = self._calculate_stats(response_times)
+        self.result['backend_metrics'] = {}
+
+        for response in self.params['results']['mw_profiler_get']:
+            start_pos = response.text.rindex('<!--')
+            end_pos = response.text.rindex('-->')
+
+            if start_pos < 0 or end_pos < 0:
+                logger.warn('Cannot find backend performance metrics')
```

Comment:
I was thinking about this but I couldn't come up with a solution which will be of any use to the user. Here we parse huuuge HTML as an text and not finding correct placeholder is the only clue we can get here. Sending this 3k+ HTML to a log is not an option.